### ISSUE TITLE
Hi guys, I am new for the MVC class. in the First Codes Migration video, Vidly mentions that there is IdentityModels class in the model, but do not have it(even I follow each his class). anyone know where I can download it?

### ISSUE THREAD

**Issue by colinjava** [mosh-hamedani/vidly-mvc-5#46]

**Comment by colinjava** [comment]
when I try to Enable-Migration from Package Management Console of GPM as video , I got below Error that  I think it caused by missed the IndentityModels.cs class in my project. 
"enable-migrations : The term 'enable-migrations' is not recognized as the name of a cmdlet, function, script file, or operable 
"

**Comment by colinjava** [comment]
anyone help me please?!

**Comment by ghardy2k** [comment]
You are in package manager console, right?

**Comment by colinjava** [comment]
Yes, I am. When I do"enable-migrations" , I got above error, that is why I think maybe I missed some classes.  
![image](https://user-images.githubusercontent.com/39657985/43347959-a4f8c15a-91c5-11e8-9f8d-4e3df78473cc.png)

**Comment by colinjava** [comment]
see I only have 2 model classes in the Model folder when I followed the Mosh video.

**Comment by colinjava** [comment]
BTW, I have the MS Aspnet Identity installed from NuGet.
![image](https://user-images.githubusercontent.com/39657985/43348110-6c229bf2-91c6-11e8-9b05-59910c894694.png)

**Comment by colinjava** [comment]
any idea?

**Comment by ghardy2k** [comment]
I don't see a routeconfig.cs file

**Comment by colinjava** [comment]
Hi ghardy2k, the routeconfig.cs is under App_Start folder with other config files on the first screen shot.

**Comment by ghardy2k** [comment]
Hmmm...try reinstall with the commmand: Install-Package EntityFramework -IncludePrerelease

**Comment by colinjava** [comment]
Thanks ghardy! I have resolved this problem by 1) reinstalled EntityFramework, IdentityFramework, and other related packages; 2) added IdentityModels.cs and; 3) corrected IdentityConfig.cs Many thanks again.

**Comment by ghardy2k** [comment]
Great!  Now that mosh taught me entity framework and mvc, I'm not sure I can go back to Web forms now.  But it was a painful transition at first. Good luck!

**Comment by colinjava** [comment]
Thanks. I hope you do well.

**Comment by 4thworldsoldier** [comment]
Hi guys i have same problem,can i know the list of packages i neee to install apart from entity frameworks and identity also how did you add identitymodels and how do i find identityconfig and what do i correct there

**Comment by jas112** [comment]
This issue can be avoided by enabling authentication when you first create the project in Visual Studio.

**Comment by agmohammed** [comment]
if any VS 2017 users encountered an error with Identity Model don't worry. you can download the identity model from https://github.com/jesselott/vidly-mosh-tutorial/blob/master/IdentityModels.cs 
and you guys need to install Microsoft.AspNet.Identity.Core and Microsoft.AspNet.EntityFrameWork from nugget package manager. Hope this helps.

**Comment by alinaer** [comment]
I had the same problem and solved it  by enabling authentication when first create the project in Visual Studio 2017, thanks for this advice to jas112.

**Comment by CDDFERREIRA** [comment]
Thanks for the help. I was following the tutorial course videos, and for someone without any idea about .ASP took me a while to find out what was wrong. I could not even enable migrations as was missing references to the datacontext, however, none of those steps were mentioned on the videos. I could not understand till know where the things came from...

**Comment by CDDFERREIRA** [comment]
I was finally able to run the "Enable-Migrations" command. 
Where does the tables mentioned on the video come from? AspNetRoles? AspNetUsers?

I dont have them from scratch...

**Comment by CDDFERREIRA** [comment]
For those that do not understand where some files show up or cannot run the Enable-Migrations command. As mentioned above, the easiest way is to create a new project and make sure Authentication is Enabled. Or, follow @agmohammed suggestion.

Also, for those using VS2017 and the last Entity Frame 6.3 there was something odd going, as it is impossible to run the command, got multiple errors, from Path Null, when definelty had it selected on dropdown, or command unknown. 
https://stackoverflow.com/questions/14410987/entity-framework-code-first-migration-error 

"Visual studio 2019: Uninstall-Package EntityFramework -Force
then
Install-Package EntityFramework -Pre -Version 6.2.0
something is up with v6.3.0"
